src/app/onion/shared/status-indicator.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit, Input } from '@angular/core';
+
```

Comment:
I opted for a single file component for this because it is in fact so tiny. In my opinion, this reduces bloat in the source code directory and makes it easier to find. We don't tend to use single file components often. What are other's thoughts on this?

Reply:
### Update

I can't find a good way to bring the nice-to-haves of SCSS (importing the variables file, functions, etc...) to the inline styles. I've extracted the styles to their own file but left the template inside the component.